packages/block-library/src/button/edit.native.js

```diff
@@ -302,48 +210,49 @@ class ButtonEdit extends Component {
 	}
 
 	dismissSheet() {
-		this.setState( {
-			isLinkSheetVisible: false,
-		} );
+		this.onHideLinkSettings();
 		this.props.closeSettingsBottomSheet();
 	}
 
-	getLinkSettings( url, rel, linkTarget, isCompatibleWithSettings ) {
+	getLinkSettings( isCompatibleWithSettings, extractSettings ) {
+		const { isLinkSheetVisible } = this.state;
+		const { attributes, setAttributes } = this.props;
+		const actions = [
+			{
+				label: __( 'Remove link' ),
+				onPress: this.onClearSettings,
+			},
+		];
+
+		const options = {
+			url: {
+				showIcon: ! isCompatibleWithSettings,
```

Comment:
I believe that the `showIcon` could be a prop that is passed directly to the LinkSettings